proposals/extensions.md

```diff
@@ -92,48 +101,56 @@ TODO should we have a naming convention like `Extension` suffixes? (`DataObjectE
 
 ## Extension type
 
-A extension type (new kind of type) is declared by a *extension_declaration*.  
-The permitted modifiers on an extension type are `partial`, 
-`unsafe`, `static`, `file` and the accessibility modifiers.  
-A static extension shall not be instantiated, shall not be used as a type and shall
-contain only static members.  
-The standard rules for modifiers apply (valid combination of access modifiers, no duplicates).  
+An extension type (new kind of type) is declared by a *extension_declaration*.  
 
 The extension type does not **inherit** members from its underlying type 
 (which may be `sealed` or a struct), but
 the lookup rules are modified to achieve a similar effect (see below).  
 
-The *extension_underlying_type* type may not be `dynamic`, a pointer, a nullable reference (no top-level nullability), 
-a ref struct type or an extension.  
-The underlying may not include an *interface_type_list* (this is part of Phase C).  
-The extension type must be static if its underlying type is static.  
+There is an identity conversion between an extension and its underlying type,
+and between an extension and its base extensions.
 
-It is a compile-time error if the underlying type differs amongst all the parts
-of an extension declaration.  
+An extension type satisfies the constraints satisfied by its underlying type (see section on constraints). 
+In phase C, some additional constraints can be satisfied (additional implemented interfaces).  
 
-It is a compile-time error if the `implicit` or `explicit` modifiers differ amongst
-all the parts of an extension declaration.  
+### Underlying type
+
+The *extension_underlying_type* type may not be `dynamic`, a pointer, 
+a ref struct type, a ref type or an extension.  
+The underlying may not include an *interface_type_list* (this is part of Phase C).  
+The extension type must be static if its underlying type is static.  
 
 When a partial extension declaration includes an underlying type specification,
 that underlying type specification shall reference the same type as all other parts
 of that partial type that include an underlying type specification.
 It is a compile-time error if no part of a partial extension includes
 an underlying type specification.  
 
+It is a compile-time error if the underlying type differs amongst all the parts
+of an extension declaration.  
+
+TODO2 we need rules to only allow an underlying type that is compatible with the base extensions.  
+
 TODO should the underlying type be inferred when none was specified but
 base extensions are specified?
 
+### Modifiers
+
+It is a compile-time error if the `implicit` or `explicit` modifiers differ amongst
+all the parts of an extension declaration.  
+
+The permitted modifiers on an extension type are `partial`, 
+`unsafe`, `static`, `file` and the accessibility modifiers.  
```

Comment:
> `file` and the accessibility modifiers

Is `file` _not_ an accessibility modifier?

(I don't know if I'm overstepping my boundary here, commenting on this PR)

Reply:
It's subtly different than an accessibility modifier. Accessibility modifiers mean that lookup finds the entity, but then it may not be accessible.
With `file`, lookup will not even find the entity that is not visible.
https://github.com/dotnet/csharplang/blob/main/proposals/csharp-11.0/file-local-types.md